pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -67,6 +73,25 @@ buildDotnetModule rec {
 
   executables = [ meta.mainProgram ];
 
+  dotnetBuildFlags =
+    let
+      # From https://nexus-mods.github.io/NexusMods.App/developers/Contributing/#for-package-maintainers
+      # TODO: add DefineConstants support to buildDotnetModule
+      constants = [
+        # Tell the app it is a distro package; affects wording in update prompts
+        "INSTALLATION_METHOD_PACKAGE_MANAGER"
+
+        # Don't include upstream's 7zz binary; we use the nixpkgs version
+        "NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR"
+      ];
+    in
+    [
+      # From https://github.com/Nexus-Mods/NexusMods.App/blob/v0.5.3/src/NexusMods.App/app.pupnet.conf#L38
+      "--property:Version=${version}"
+      "--property:TieredCompilation=true"
```

Comment:
You'll also want the `ReadyToRun` flag here; it improves startup times, especially in something as JIT heavy at startup as NMA.

Reply:
We removed ReadyToRun because we didn't have crossgen2 support in nix before, I believe after the changes by @corngood that might've been solved though.

Reply:
Added in 0205410716648086aa4343996857bb1a3ef42cb3. I'll leave this thread unresolved until we can identify if it a) works with `dotnet build` (I can only find it documented for `dotnet publish`) and b) doesn't introduce issues related to crossgen2.

Reply:
Note @GGG-KILLER previously advised against this (https://github.com/NixOS/nixpkgs/pull/318647#discussion_r1647471954):
> This makes it so that the code is pre-compiled to the target runtime, given that we don't add the framework's RID unless it's a self-contained build, I don't recommend setting this.



...but the nexus folks claim it has some performance benefits.

I'm not sure what is meant by _"framework's RID"_ so I'm not in a position to make an educated decision.